src/routers/rsvp.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express')
+const RSVP = require('../models/RSVPModel')
+const router = new express.Router()
+const authMiddleware = require('../middleware/authMiddleware')
+
+router.post('/rsvp', authMiddleware, async (req, res) => {
+  try {
+    const rsvp = new RSVP(req.body)
+    rsvp.owner = req.user._id
+    const existingRSVP = await RSVP.findOne({ owner: req.user._id })
+    if (existingRSVP) {
+      throw new Error('This user has already RSVP\'d')
+    }
+    await rsvp.save()
+    res.status(201).send(rsvp)
+  } catch (err) {
+    res.status(400).send({ errorMessage: err.message })
+  }
+})
+
+router.get('/rsvp/', authMiddleware, async (req, res) => {
```

Comment:
/rsvp/:id is more concise, Change it to that implementation. I remember talking about it doesnt make sense for us to pull a specific RSVP but just to be consistent with other routes/models, change the implementation to reflect that.